### ISSUE TITLE
Fix MSBT

### ISSUE THREAD

**Issue by Le0X8** [acridotheres/3ds-formats#1]
There are still some errors with parsing some MSBT files. I iterated over all the dialogue files in ACNL, logs are attached and very long, maybe add some test cases for these files.

Used command (after extracting all umsbt files):

```sh
for file in */*.msbt
do
echo $file
acr x -f msbt -i $file
done &> out.log
```

[out.log](https://github.com/user-attachments/files/17894706/out.log)

**Comment by Le0X8** [comment]
After reading 15,5k lines of logs, here's a list with all affected files:

- AN_3P_Fu
- AN_Dream_Field
- AN_Dt_Club
- AN_Dt_Depart
- AN_Ev_PcbdayH
- AN_Ev_Xmas_Hint
- AN_FreeA_Always
- AN_FreeA_Friend
- AN_FreeF_Newyear
- AN_Q07_Req
- AN_Q08_Reserve
- AN_Q09_Reserve
- AN_Sp_Phouse
- AN_Sp_Prologue
- BO_Ai_Rain2
- BO_AiV_Snow2
- BO_Ev_PcbdayH
- BO_Ev_Xmas_Hint
- BO_FreeB_Obj
- BO_FreeG_Visitor
- BO_Q06_A_Timeover
- BO_Q07_Req
- BO_Q08_Furniture
- BO_Q08_Reserve
- BO_Q09_Reserve
- BO_Re_Newyear
- BO_Sp_Museum_Art
- BO_Sp_Museum_Fish
- BO_Sp_Museum_Fossil
- BO_Sp_Museum_Insect
- BO_Sp_Phouse
- FU_Ai_House3
- FU_AiV_Fine2
- FU_Ev_PcbdayH
- FU_Ev_Xmas_Hint
- FU_FreeA_Always
- FU_FreeF_0229
- FU_Q06_A_Timeover
- FU_Q07_Req
- FU_Q09_Reserve
- FU_Re_Beeface
- FU_Re_FirstV
- FU_Sp_Phouse
- GE_3P_Ot
- GE_AiV_Snow2
- GE_Ev_PcbdayH
- GE_Ev_Xmas_Hint
- GE_FreeA_RumorN
- GE_FreeB_ItemP
- GE_Q06_A_Timeover
- GE_Q07_Req
- GE_Q08_Reserve
- GE_Q09_Reserve
- GE_Re_Beeface
- GE_Re_Napping
- GE_Sp_Camp_Game1
- GE_Sp_Phouse
- HA_3P_An
- HA_Ai_Rain2
- HA_Ev_PcbdayH
- HA_Ev_Xmas_Hint
- HA_FreeA_Rumor
- HA_Q06_A_Timeover
- HA_Q07_Req
- HA_Q08_Furniture
- HA_Q08_Reserve
- HA_Q09_Reserve
- HA_Sp_Museum_Art
- HA_Sp_Museum_Fish
- HA_Sp_Museum_Fossil
- HA_Sp_Museum_Insect
- HA_Sp_Phouse
- KO_3P_Bo
- KO_Ai_Snow2
- KO_AiV_Fine2
- KO_Ev_PcbdayH
- KO_Ev_Xmas_Hint
- KO_Q06_A_Timeover
- KO_Q07_Req
- KO_Q08_Reserve
- KO_Q09_Reserve
- KO_Sp_Phouse
- NPC_Cafe_KappeiM
- NPC_Cafe_Kotobuki
- NPC_CafeOrder_Secretary
- NPC_Event_Newyear
- NPC_Howto_2
- NPC_Howto_3
- NPC_KameyamaA
- NPC_Kappei_Mom
- NPC_Palloncino_Badge
- NPC_Prologue_1
- OT_AiV_Fine2
- OT_AiV_Snow1
- OT_Ev_Xmas_Hint
- OT_FreeB_ItemP
- OT_FreeG_Host
- OT_Q06_A_Timeover
- OT_Q07_Req
- OT_Q08_Reserve
- OT_Q09_Reserve
- OT_Sp_Phouse
- SYS_Get_Badge
- SYS_Motto
- SYS_Museum_Fish
- SYS_Museum_Insect
- SYS_Start
- ZK_3P_An
- ZK_3P_Ge
- ZK_Ai_Snow1
- ZK_AiV_Snow1
- ZK_Ev_Xmas_Hint
- ZK_FreeB_ItemP
- ZK_FreeF_0229
- ZK_FreeG_Host
- ZK_Q06_A_Timeover
- ZK_Q07_Req
- ZK_Q08_Reserve
- ZK_Q09_Reserve
- ZK_Re_7days
- ZK_Re_Poison
- ZK_Sp_Phouse
- ZK_Sp_Prologue

Most of them are "invalid section" errors, but some aren't and have to be investigated further.

**Comment by Le0X8** [comment]
most of the errors should be fixed now in v0.2.1

If the error persists, I'll open a new issue as most of the errors were caused by a bug in the padding calculation function.

**Comment by Le0X8** [comment]
Yes. Everything is fixed now, there are no errors anymore :D
